src/vs/workbench/browser/parts/quickopen/quickopen.ts

```diff
@@ -22,15 +22,15 @@ export class GlobalQuickOpenAction extends Action {
 	public static ID = 'workbench.action.quickOpen';
 	public static LABEL = nls.localize('quickOpen', "Go to File...");
 
-	constructor(id: string, label: string, @IQuickOpenService private quickOpenService: IQuickOpenService) {
+	constructor(id: string, label: string, @IQuickOpenService private quickOpenService: IQuickOpenService, @ICommandService private commandService: ICommandService) {
 		super(id, label);
 
 		this.order = 100; // Allow other actions to position before or after
 		this.class = 'quickopen';
 	}
 
 	public run(): TPromise<any> {
-		this.quickOpenService.show(null);
+		this.commandService.executeCommand('vscode.quickOpen');
```

Comment:
This is madness. Don't add a new command, replace the action with a command having the same id, register the command to the command palette. Don't add an API command